### PR TITLE
Support DensityInterface v0.4

### PR DESCRIPTION
**Pull request by oschulz**
This is non-breaking and keeps support for DensityInterface v0.3.2.

### CODE REVIEW

**Review by devmotion — COMMENTED** [review]
Thanks for the PR! Is there a good reason to keep support for < 0.4? I would assume it is easier to just switch to 0.4.

Can you also update the version number of Distributions?

**Comment by oschulz** [comment]
> I would assume it is easier to just switch to 0.4.

I guess so, not many packages depend on DensityInterface so far.

Ok, dropped DensityInterface v0.3 and updated Distributions version number.

File: src/density_interface.jl [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@inline DensityInterface.hasdensity(::Distribution) = true
+@inline DensityInterface.DensityKind(::Distribution) = HasDensity()
```
Comment (devmotion) [inline src/density_interface.jl:1]: I think it's not exported, is it?
```suggestion
@inline DensityInterface.DensityKind(::Distribution) = DensityInterface.HasDensity()
```

Comment (oschulz) [reply src/density_interface.jl:1]: Hm, it actually *is* exported:

https://github.com/JuliaMath/DensityInterface.jl/blob/e71eb2ded6f3320d66a6d7efc615e3e60dddb58f/src/interface.jl#L69

Comment (oschulz) [reply src/density_interface.jl:1]: Ah, but we "import DensityInterface" :-)

**Review by devmotion — APPROVED** [review]
Thank you!

**Comment by oschulz** [comment]
Thanks!

**Pull request merged**